connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/Plugins.java

```diff
@@ -71,10 +71,23 @@ private static DelegatingClassLoader newDelegatingClassLoader(final List<String>
         return Utils.join(plugins, ", ");
     }
 
-    protected static <T> T newPlugin(Class<T> klass) {
+    protected <T> T newPlugin(Class<T> klass) {
+        ClassLoader classLoader = delegatingLoader.pluginClassLoader(klass.getName());
+        if (classLoader == null) {
+            classLoader = delegatingLoader;
+        }
+        return newPlugin(classLoader, klass);
+    }
+
+    protected static <T> T newPlugin(ClassLoader classLoader, Class<T> klass) {
+        ClassLoader oldClassLoader = compareAndSwapLoaders(classLoader);
```

Comment:
nit: let's keep calling this `savedLoader` as in the other places where we call this method because: a) naming consistency might pay off here more than elsewhere, b) calling it old might be misleading because you are keeping the reference because you want to restore it somewhere (rather than replace it as "old" would imply)